internal/backend/runtime/omni/controllers/omni/machine_extensions.go

```diff
@@ -213,9 +226,65 @@ func (ctrl *MachineExtensionsController) Reconcile(ctx context.Context,
 	return tracker.cleanup(ctx)
 }
 
-func (ctrl *MachineExtensionsController) getRelatedClusterMachines(ctx context.Context,
-	r controller.QRuntime, configuration *omni.ExtensionsConfiguration,
-) ([]*omni.ClusterMachine, error) {
+// determineExtensions determines the extensions for the given cluster machine.
+// The extensions are determined in the following order:
+// 1. Extensions defined for the cluster machine itself.
+// 2. Extensions defined for the machine set the cluster machine belongs to.
+// 3. Extensions defined for the cluster machine belongs to.
+//
+// If there are multiple extensions defined for the same level, the one with the lexicographically highest ID is used.
+func (ctrl *MachineExtensionsController) determineExtensions(cm *omni.ClusterMachine, configs iter.Seq[*omni.ExtensionsConfiguration]) []string {
```

Comment:
also as this has something I think is counter-intuitive - I would expect probably extensions to be merged, not replaced (I guess this matches config patches more closely)